### DISCUSSION TITLE
Is it okay to use imperative method such as `flyTo` in a controlled map?

### DISCUSSION THREAD

**Discussion by milhamm** [uber/react-map-gl#2294] [Q&A] [answered]
I'm using a controlled map where I passed in the view state via props. In my case, I want to achieve a change in view position with some animation like the behavior of a `flyTo` method.

I just wondering if it okay to use imperative method such as `flyTo` with a controlled map. I have read the limitation that using an imperative method  in a controlled map will make the map's state to deviate from its props.

**Comment by Pessimistress** [comment, marked as answer]
If you call `flyTo` on a controlled map, `onMove` will be invoked as the map attempts to animate its camera. Your app is responsible of listening to the event and update the props.

https://visgl.github.io/react-map-gl/docs/api-reference/map#onmove

**Comment by milhamm** [comment]
Got it @Pessimistress 
Thanks
